client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -187,16 +187,23 @@ boolean handleEndOfSegment(SegmentWithRange segmentCompleted) throws ReaderNotIn
         AtomicBoolean reinitRequired = new AtomicBoolean(false);
         boolean result = sync.updateState((state, updates) -> {
             if (!state.isReaderOnline(readerId)) {
+                log.error("Reader " + readerId + " is offline according to the state but is attempting to update it.");
```

Comment:
I believe this should be a `warn`. No actual error got caught, but it is an invalid/unexpected condition.

Reply:
Our definition of error is "Something bad happened that can't be automatically be handled"
It meets this criteria. If this line is reached a user will have to reinitialize the reader because we cannot get it back into a good state.

Reply:
OK. Got it.